### ISSUE TITLE
Solargraph Request textDocument/hover failed

### ISSUE THREAD

**Issue by rhymes** [castwide/vscode-solargraph#53]
I can't get solargraph to work at all :( I'm getting this error:

```text
[Error - 4:06:03 PM] Request textDocument/hover failed.
  Message: [TypeError] no implicit conversion of Solargraph::Workspace into String
  Code: -32603 
```

solargraph 0.21.0
ruby 2.5.1
VSCode 1.23.0

**Comment by castwide** [comment]
There appear to be problems running the Solargraph gem with Ruby 2.5. Tests have been updated to include it. Issues with 2.5 are being tracked at castwide/solargraph#55.

**Comment by castwide** [comment]
@rhymes Are you using the Runtime plugin? i.e., this configuration in a .solargraph.yml file:

```
plugins:
- runtime
```

The Runtime plugin is still experimental, and I recently found a bug in it that might cause the Solargraph::Workspace error you're seeing.

**Comment by rhymes** [comment]
@castwide you were right, I removed that line and I stopped seeing the error. Still it won't work with Ruby 2.5. I'll wait for the next compatible release then

**Comment by castwide** [comment]
One more thing you can try. The gem ships with core documentation for Ruby 2.2.2. You can update to a newer version by running `solargraph download-core`. That should get you the documentation for the closest version it can find, which is currently 2.4.2...not a perfect match, but it might help. A download for 2.5.1 documentation is forthcoming.

**Comment by castwide** [comment]
Gem version 0.21.1 is published with fixes for Ruby 2.5.1 compatibility.

**Comment by rhymes** [comment]
Seems to be working for me @castwide and it's awesome! Thank you!
